x509-cert/src/attr.rs

```diff
@@ -229,7 +229,7 @@ impl FromStr for AttributeTypeAndValue {
 
         // Either decode or lookup the OID for the given key.
         let oid = match DB.by_name(key) {
-            Some(oid) => *oid,
+            Some(oid) => oid.try_into().unwrap(),
```

Comment:
Overall I don't know how I feel about this move.

One of the issues I have is this conversion. I feel like I should be able to `impl From<ObjectIdentifierRef> for ObjectIdentifier` and get this out of the way.
Even if it relies on an unsafe conversion underneath.
If you created an `ObjectIdentifierRef` then obviously the content is valid already.

Any opinion about a:
```
impl From<&ObjectIdentifierRef> for ObjectIdentifier<DEFAULT_MAX_SIZE>
```
(specifically `DEFAULT_MAX_SIZE` as this is supposed to be the maximum length of the underlying slice)

Reply:
`ObjectIdentifierRef` isn't bounded by `DEFAULT_MAX_SIZE`, so this conversion would potentially have to panic if it exceeded `DEFAULT_MAX_SIZE`.

I'm also not sure it's a good idea to place restrictions on `ObjectIdentifierRef` like that: most of the current unreleased changes are trying to make the crate more flexible around handling OIDs which are larger than the default.